pkg/proctree/exit_processing.go

```diff
@@ -0,0 +1,96 @@
+package proctree
+
+import (
+	"fmt"
+
+	"github.com/aquasecurity/tracee/signatures/helpers"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// processExitEvent remove references of processes from the tree when the corresponding process exit without children, or
+// if the last child process of a process exits.
+// Notice that there is a danger of memory leak if there are lost events of sched_process_exit (but is limited to the
+// possible number of PIDs - 32768)
+func (tree *ProcessTree) processExitEvent(event *trace.Event) error {
+	err := tree.processDefaultEvent(event)
+	if err != nil {
+		return err
+	}
+	process, err := tree.getProcess(event.HostProcessID)
+	if err != nil {
+		return fmt.Errorf("process was inserted to the treee but is missing right after")
+	}
+	process.Mutex.Lock()
+	defer process.Mutex.Unlock()
+	thread, ok := process.Threads.Get(event.HostThreadID)
+	if !ok {
+		thread = &threadInfo{}
+	}
+	thread.exitTime = timestamp(event.Timestamp)
+	process.Threads.Set(event.HostThreadID, thread)
+
+	processGroupExit, err := helpers.GetTraceeBoolArgumentByName(*event, "process_group_exit")
+	if err != nil {
+		return err
+	}
+
+	if processGroupExit {
+		process.IsAlive = false
+		process.ExitTime = timestamp(event.Timestamp)
+		for _, tid := range process.Threads.Keys() {
```

Comment:
Wouldn't it be better to range on `process.Threads.Values()`? Or are those copies? If the latter maybe we should rethink having the `Values()` function or add a helper which gets an array with the references directly for modification, because this seems to be about roundabout currently.